app/screens/map/map_screen.tsx

```diff
@@ -66,7 +75,9 @@ export default function MapScreen({
       }
     }
 
-    getCurrentLocation();
+    if (firstLocation === undefined) {
+      getCurrentLocation();
+    }
```

Comment:
Why not call the location parameter of the route inside this useEffect instead of a const outside ?

Reply:
This was made in case I needed the `firstLocation` somewhere else, to make it more robust.